src/components/EditionCard/DownloadLink.tsx

```diff
@@ -1,20 +1,29 @@
 import { Box, Icon } from "@nypl/design-system-react-components";
+import { useRouter } from "next/router";
 import React from "react";
 import Link from "~/src/components/Link/Link";
 import { trackCtaClick } from "~/src/lib/adobe/Analytics";
+import { fulfillFetcher } from "~/src/lib/api/SearchApi";
 import { ItemLink } from "~/src/types/DataModel";
 import { formatUrl } from "~/src/util/Util";
 
 const DownloadLink: React.FC<{
   downloadLink: ItemLink;
   title: string;
   isLoggedIn: boolean;
-}> = ({ downloadLink, title, isLoggedIn }) => {
+  loginCookie?: string;
+}> = ({ downloadLink, title, isLoggedIn, loginCookie }) => {
+  const router = useRouter();
   if (downloadLink && downloadLink.url) {
     let linkText = "Download PDF";
     let linkUrl = formatUrl(downloadLink.url);
 
-    const trackDownloadCta = () => {
+    const handleOnClick = async (e) => {
```

Comment:
Let's rename this to something more descriptive like `handleDownload` or `handleFulfillmentDownload` -- not sure what makes the most sense here.

Reply:
ohh i see this is used for multiple different types of downloading/viewing content. so maybe `handleDownload` is best? or maybe `handleOnClick` is ok since it is generic.